R/recode-with-table.R

```diff
@@ -580,170 +595,162 @@ get_data_variable_name <-
 #' @keywords internal
 recode_columns <-
   function(data,
-           variables_details_rows_to_process,
+           details,
            data_name,
            log,
            print_note,
            else_default,
            tables) {
-    # Split variables to process into recode map and func
-    map_variables_to_process <-
-      variables_details_rows_to_process[grepl(pkg.env$recode.key.map, variables_details_rows_to_process[[pkg.env$columns.recTo]]),]
-    if (nrow(map_variables_to_process) > 0) {
-      stop(paste0(pkg.env$recode.key.map, "variables were detected however this feature is not yet supported"))
+
+    var_col <- pkg.env$columns.variable
+    varstart_col <- pkg.env$columns.variableStart
+    recto_col <- pkg.env$columns.recTo
+
+    builtin_functions <- c(
+      pkg.env$recode.key.func,
+      pkg.env$recode.key.map,
+      pkg.env$recode.key.id.from)
+    builtin_function_regex <- paste0(builtin_functions, collapse = "|")
+
+    map_vars <- details[
+      grepl(pkg.env$recode.key.map, details[[recto_col]]), ]
+    if (nrow(map_vars) > 0) {
+      stop(
+        paste(
+          pkg.env$recode.key.map,
+          "variables were detected, however,",
+          "this feature is not yet supported"))
     }
 
-    id_variables_to_process <-
-      variables_details_rows_to_process[grepl(pkg.env$recode.key.id.from, variables_details_rows_to_process[[pkg.env$columns.recTo]]),]
-
-    func_variables_to_process <-
-      variables_details_rows_to_process[grepl(pkg.env$recode.key.func, variables_details_rows_to_process[[pkg.env$columns.recTo]]),]
-
-    non_derived_keys <- c(pkg.env$recode.key.func,pkg.env$recode.key.map,pkg.env$recode.key.id.from)
-    # These are all the variables which do not need a function to be recoded
-    rec_variables_to_process <- variables_details_rows_to_process
-    for(variable in unique(variables_details_rows_to_process$variable)) {
-      if(sum(grepl(
-        paste(non_derived_keys, collapse = "|"),
-        rec_variables_to_process[rec_variables_to_process$variable == variable, "recEnd"]
-      )) != 0) {
-        rec_variables_to_process <- rec_variables_to_process[rec_variables_to_process$variable != variable, ]
+    idfrom_vars <- details[
+      grepl(pkg.env$recode.key.id.from, details[[recto_col]]), ]
+
+    func_vars <- details[
+      grepl(pkg.env$recode.key.func, details[[recto_col]]), ]
+
+    # These are all the variables that don't need builtin functions to be
+    # recoded
+    normal_vars <- details
```

Comment:
I assume that this variable replaces the deleted `rec_variables_to_process` variable on line 604. If so, this also includes user defined functions right whereas the deleted variable does not.